metadata-io/src/main/java/com/linkedin/metadata/graph/elastic/ElasticSearchGraphService.java

```diff
@@ -0,0 +1,213 @@
+        relationshipFilter,
+        offset,
+        count
+    );
+
+    return Arrays.stream(response.getHits().getHits())
+        .map(hit -> ((HashMap<String, String>) hit.getSourceAsMap().getOrDefault(destinationNode, EMPTY_HASH)).getOrDefault("urn", null))
+        .filter(urn -> urn != null)
+        .collect(Collectors.toList());
+  }
+
+  private Filter createUrnFilter(@Nonnull final Urn urn) {
+    Filter filter = new Filter();
+    CriterionArray criterionArray = new CriterionArray();
+    Criterion criterion = new Criterion();
+    criterion.setCondition(Condition.EQUAL);
+    criterion.setField("urn");
+    criterion.setValue(urn.toString());
+    criterionArray.add(criterion);
+    filter.setCriteria(criterionArray);
+
+    return filter;
+  }
+
+  public void removeNode(@Nonnull final Urn urn) {
+    Filter urnFilter = createUrnFilter(urn);
+    Filter emptyFilter = new Filter().setCriteria(new CriterionArray());
+    List<String> relationshipTypes = new ArrayList<>();
+
+    RelationshipFilter outgoingFilter = new RelationshipFilter().setDirection(RelationshipDirection.OUTGOING);
+    RelationshipFilter incomingFilter = new RelationshipFilter().setDirection(RelationshipDirection.INCOMING);
+
+    _graphReadDAO.deleteByQuery(
+        null,
+        urnFilter,
+        null,
+        emptyFilter,
+        relationshipTypes,
+        outgoingFilter
+    );
+
+    _graphReadDAO.deleteByQuery(
+        null,
+        urnFilter,
+        null,
+        emptyFilter,
+        relationshipTypes,
+        incomingFilter
+    );
+
+    return;
+  }
+
+  public void removeEdgeTypesFromNode(
+      @Nonnull final Urn urn,
+      @Nonnull final List<String> relationshipTypes,
+      @Nonnull final RelationshipFilter relationshipFilter) {
+
+    Filter urnFilter = createUrnFilter(urn);
+    Filter emptyFilter = new Filter().setCriteria(new CriterionArray());
+
+    _graphReadDAO.deleteByQuery(
+        null,
+        urnFilter,
+        null,
+        emptyFilter,
+        relationshipTypes,
+        relationshipFilter
+    );
+  }
+
+  @Override
+  public void configure() {
+    log.info("Setting up elastic graph index");
+    boolean exists = false;
+    try {
+      exists = searchClient.indices().exists(
+          new GetIndexRequest(_indexConvention.getIndexName(INDEX_NAME)), RequestOptions.DEFAULT);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    // If index doesn't exist, create index
+    if (!exists) {
+      log.info("Elastic Graph Index does not exist. Creating.");
+      CreateIndexRequest createIndexRequest = new CreateIndexRequest(_indexConvention.getIndexName(INDEX_NAME));
+
+      createIndexRequest.mapping(GraphRelationshipMappingsBuilder.getMappings());
+      createIndexRequest.settings(SettingsBuilder.getSettings());
+
+      try {
+        searchClient.indices().create(createIndexRequest, RequestOptions.DEFAULT);
+      } catch (IOException e) {
+        e.printStackTrace();
```

Comment:
log? i we should throw or something here i think.

Reply:
and added logs here too